engine/movie/oak_speech/oak_speech.asm

```diff
@@ -49,15 +57,18 @@ OakSpeech:
 	ld [wcf91], a
 	ld a, 1
 	ld [wItemQuantity], a
-	call AddItemToInventory  ; give one potion
+	call AddItemToInventory
 	ld a, [wDefaultMap]
 	ld [wDestinationMap], a
+	; This call is required for running StartNewGameDebug properly.
+	; StartNewGame also needs it to land the player outside their
+	; house when they warp on the mat after this routine ends.
```

Comment:
...In other words, this line is required for the game to work properly, just like 99% of all lines? What is this comment clarifying?

Reply:
It wasn't at all clear that the game uses a special warp when starting a debug and a non-debug new game, especially with the old labels.

```suggestion
	; In debug mode, land the player in wLastMap (PALLET_TOWN).
	; In non-debug mode, land the player in their room (REDS_HOUSE_2F).
```

Reply:
I still don't see the issue. This is the routine for Oak's intro speech before a new game, and it clearly calls `SpecialWarpIn`, the routine responsible for setting up special warps, including the one to the initial map (which is `PALLET_TOWN` in debug or else `REDS_HOUSE_2F`). If you think the name "`SpecialWarpIn`" doesn't make that clear (and I agree, I can't tell what the verb+noun are in it), it's what needs renaming, not commenting in this one usage of it.

Reply:
What would be a good new name for the `SpecialWarpIn` routine?

Reply:
`PrepareForSpecialWarp`?

Reply:
That will be its new name then 😄